core.js

```diff
@@ -0,0 +1,685 @@
+   *  //for each specifying route
+   *  for(routeI = fromConnection.routes.length - 1;
+   *      routeI >= 0;
+   *      routeI--){
+   *    route = fromConnection.routes[routeI];
+   *    //unregister this connection from this route
+   *    // OH! actually connections are not registered with routes
+   *  }*/
+  sourceConnection.routes = [];
+};
+
+/**
+ * cleans up both sides of all connections associated with
+ *   the provided set of endpoints.
+ * @param {Array} fromEndpoints an array of publishers or subscribers
+ *   to clean up
+ */
+Manager.cleanConnectionsFrom = function(fromEndpoints){
+  var fromEndpoint,
+      fromEndpointI,
+      fromConnection,
+      fromConnectI;
+
+  //for each endpoint
+  for(fromEndpointI = fromEndpoints.length - 1;
+      fromEndpointI >= 0;
+      fromEndpointI--){
+    fromEndpoint = fromEndpoints[fromEndpointI];
+    //for each connection this endpoint is involved in
+    for(fromConnectI = fromEndpoint.connectedTo.length - 1;
+        fromConnectI >= 0;
+        fromConnectI--){
+      fromConnection = fromEndpoint.connectedTo[fromConnectI];
+      //break the connection
+      Manager.breakConnection(fromEndpoint, fromConnection);
+//      //break the connection from the other side too
+//      toEndpoint = fromConnection.endpoint;
+//      var cleanedTo = false;
+//      //for each connection the other endpoint is involved in
+//      for(toConnectI = toEndpoint.connectedTo.length -1;
+//          toConnectI >= 0 && !cleanedTo;
+//          toConnectI--){
+//        toConnection = toEndpoint.connectedTo[toConnectI];
+//        //if the OTHER other endpoint matches the fromEndpoint
+//        if (toConnection.endpoint === fromEndpoint){
+//          //then remove the connection
+//          toEndpoint.connectedTo.splice(toConnectI, 1);
+//          //there should be only one occurance
+//          cleanedTo = true;
+//        }
+//      }
+//      //if we didn't find the connection in the other endpoint,
+//      if (!cleanedTo){
+//        //TODO: log warning?
+//      }
+//      /*
+//       * We don't need to clean up each specifying route
+//       * since routes don't track connections, just matched publishers
+//       *  //for each specifying route
+//       *  for(routeI = fromConnection.routes.length - 1;
+//       *      routeI >= 0;
+//       *      routeI--){
+//       *    route = fromConnection.routes[routeI];
+//       *    //unregister this connection from this route
+//       *    // OH! actually connections are not registered with routes
+//       *  }*/
+//      fromConnection.routes = [];
+    }
+    fromEndpoint.connectedTo = [];
+  }
+};
+
+/**
+ * Removes the specified client from the server an cleans up
+ *   any associated pub/sub connections.
+ * @param client {Object|string|Leaf} details about the client.
+ *   {name:string, metadata:Object}
+ * @returns {boolean} true iff a client was removed.
+ */
+Manager.prototype.removeClient = function(client){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function 'anonymous' has a complexity of 7.](https://www.codacy.com/app/quinkennedy/spacebrew/file/2279533613/issues/source?bid=3184377&fileBranchId=3187330#l358)